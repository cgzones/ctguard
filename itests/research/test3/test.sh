#!/bin/sh

set -eu

cleanup () {
  rm -f logscan.state input.log alerts.log research.sock intervention.log
}

cleanup

touch input.log

../../../ctguard-research --cfg-file research.conf -f -x &
research_pid=$!
echo "research daemon running with pid ${research_pid}\n"

sleep 1

../../../ctguard-logscan --cfg-file logscan.conf -f -x &
logscan_pid=$!
echo "logscan daemon running with pid ${logscan_pid}\n"

sleep 1

echo "test" >> input.log
echo "test2" >> input.log

sleep 1

echo "test" >> input.log

sleep 4

logscan_pid2=`pidof ctguard-logscan`
if [ "X${logscan_pid}" != "X${logscan_pid2}" ]; then
  echo "logscan daemon not running anymore!!\n'${logscan_pid}' vs '${logscan_pid2}'\nFAILURE!!\n\n";
  exit 1
fi

kill -INT ${logscan_pid}

research_pid2=`pidof ctguard-research`
if [ "X${research_pid}" != "X${research_pid2}" ]; then
  echo "research daemon not running anymore!!\n'${research_pid}' vs '${research_pid2}'\nFAILURE!!\n\n";
  exit 1
fi

kill -INT ${research_pid}

sleep 2

echo "Comapring expected vs actual output:"
diff -u test.output.expected alerts.log

cleanup

echo "\n\n\nsuccess!!!\n\n"
