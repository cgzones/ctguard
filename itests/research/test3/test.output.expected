ALERT START
Priority:  4
Info:      test rule [1]
Log:       test
Traits:
                        control : false
                         format : unknown
                       hostname : unittest
                  source_domain : input.log
                 source_program : ctguard-logscan
                   time_scanned : Thu Jan  1 00:00:00 1970 UTC
                      time_send : Thu Jan  1 00:00:01 1970 UTC
ALERT END

ALERT START
Priority:  4
Info:      test rule [1]
Log:       test2
Traits:
                        control : false
                         format : unknown
                       hostname : unittest
                  source_domain : input.log
                 source_program : ctguard-logscan
                   time_scanned : Thu Jan  1 00:00:00 1970 UTC
                      time_send : Thu Jan  1 00:00:01 1970 UTC
ALERT END

ALERT START
Priority:  4
Info:      test rule [1]
Log:       test
Traits:
                        control : false
                         format : unknown
                       hostname : unittest
                  source_domain : input.log
                 source_program : ctguard-logscan
                   time_scanned : Thu Jan  1 00:00:00 1970 UTC
                      time_send : Thu Jan  1 00:00:01 1970 UTC
ALERT END

ALERT START
Priority:  8
Info:      No test2 after test [3]
Log:       test
Traits:
                        control : false
                         format : unknown
                       hostname : unittest
                  source_domain : input.log
                 source_program : ctguard-logscan
                   time_scanned : Thu Jan  1 00:00:00 1970 UTC
                      time_send : Thu Jan  1 00:00:01 1970 UTC
ALERT END

